src/mame/drivers/p2000t.cpp

```diff
@@ -60,6 +119,26 @@ void p2000t_state::p2000t_mem(address_map &map)
 	map(0xe000, 0xffff).bankrw(m_bank);
 }
 
+void p2000h_state::p2000h_mem(address_map &map)
+{
+	map(0x0000, 0xffff).rw(FUNC(p2000h_state::memory_read), FUNC(p2000h_state::memory_write));
+}
+
+u8 p2000h_state::memory_read(offs_t offset)
+{
+	/* if port c0-cf bit 0 is set mem page 0 (0x0000-0x1fff) is addressed as rom on reading else as RAM */
+	if ((offset < 0x2000) && m_hiresmem_bank0_ROM)
+	{
+		return m_hiresrom[offset];
+	}
+	return m_hiresram->read(offset);
+}
+void p2000h_state::memory_write(offs_t offset, u8 data)
+{
+	/* on writing it is always RAM */
+	m_hiresram->write(offset, data);
+}
```

Comment:
This should use memory banks or views to avoid going through trampolines on every access.

Reply:
Can you point out an example, IMO this is the most execution efficient implementation